pkg/webui/console/components/device-data-form/index.js

```diff
@@ -279,28 +243,34 @@ class DeviceDataForm extends Component {
           component={Input}
           disabled={external_js}
         />
-        <Form.Field
-          title={sharedMessages.appKey}
-          name="root_keys.app_key.key"
-          type="byte"
-          min={16}
-          max={16}
-          placeholder={rootKeyPlaceholder}
-          description={m.appKeyDescription}
-          component={Input}
-          disabled={external_js || rootKeysNotExposed}
-        />
-        <Form.Field
-          title={sharedMessages.nwkKey}
-          name="root_keys.nwk_key.key"
-          type="byte"
-          min={16}
-          max={16}
-          placeholder={rootKeyPlaceholder}
-          description={m.nwkKeyDescription}
-          component={Input}
-          disabled={external_js || rootKeysNotExposed}
-        />
+        {mayEditKeys && (
+          <>
```

Comment:
Please refactor the rest of the files fragments also to shorthands or leave it as `<React.Fragment />`.